include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,555 @@
+    //!\endcond
+    {
+        get_minimiser(skip);
+        return *this;
+    }
+
+    /*!\brief Forward copy of this iterator.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    window_iterator operator+(difference_type const skip) const noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_iterator tmp{*this};
+        return tmp += skip;
+    }
+
+    /*!\brief Non-member operator+ delegates to non-friend operator+.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    friend window_iterator operator+(difference_type const skip, window_iterator const & it) noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        return it + skip;
+    }
+
+    /*!\brief Move the iterator by a given offset and return the corresponding minimiser.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    value_type operator[](difference_type const n)
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_left += n;
+        get_minimiser();
+        return operator*();
+    }
+
+private:
+    //!brief Iterator to last element in range.
+    sentinel_t last_elem;
+
+    //!brief Keeps track if the minimiser value changed.
+    bool minimiser_changed{false};
+
+    //!\brief The minimiser value.
+    size_t minimiser_value{0};
+
+    //!\brief Iterator to the leftmost element of one window.
+    it_t window_left;
+
+    //!\brief Iterator to the rightmost element of one window.
+    it_t window_right;
+
+    //!\brief The number of elements in one window.
+    uint32_t num_w_elems;
+
+    //!\brief Stored values per window.
+    std::deque<uint64_t> window_values;
+
+    //!\brief Increments iterator by 1.
+    void get_minimiser()
+    {
+        if (window_values.size() == 0)
+            window_first();
+        else
+        {
+            // Call next_minimiser until minimiser value changed or end of the underlying range is reached.
+            while((!minimiser_changed) && (std::ranges::next(window_right) != last_elem))
+            {
+                next_minimiser();
+            }
+            if (!minimiser_changed)
+            {
+                std::ranges::advance(window_left, 1);
+                std::ranges::advance(window_right, 1);
+            }
+        }
+
+        minimiser_changed = false;
+    }
+
+    /*!\brief Increments iterator by `skip`.
+     * \param skip Amount to increment.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    void get_minimiser(difference_type const skip)
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        std::ranges::advance(window_left, skip);
+        window_first();
```

Comment:
I don't think you can model a random-access range. Sorry if I was not clear about this but I thought we agreed on a forward range being fine. 
The reason is that you now jump to the `skip`-th position in the underlying view but not the `skip`-th minimizer.
And you cannot know how many minimizers there are beforehand.